# a. Build 99% Confidence Interval Using Sample Standard Deviation

### Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

Sample size (n) = 15

Sample data (durability in millions of characters):
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

Confidence level = 99%
                
Since the population standard deviation is unknown, we use the sample standard deviation (s) and apply the t-distribution instead of the normal (z) distribution

In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
# Given data
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
data

array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2 ,
       1.33, 1.18, 1.22, 1.29])

In [3]:
# Sample size
n = len(data)
n

15

In [4]:
# Sample mean
mean_x = np.mean(data)
mean_x

1.2386666666666666

In [5]:
# Sample standard deviation (using n-1 for sample std deviation)
s = np.std(data, ddof=1)
s

0.19316412956959936

In [6]:
# Degrees of freedom
df = n - 1
df

14

In [7]:
# t-critical value for 99% confidence interval
t_critical = stats.t.ppf(0.995, df)
t_critical

2.976842734370834

In [8]:
# Margin of error
ME = t_critical * (s / np.sqrt(n))
ME

0.14846932822817596

In [9]:
# Confidence Interval
CI_lower = mean_x - ME
CI_upper = mean_x + ME
(CI_lower, CI_upper)

(1.0901973384384906, 1.3871359948948425)

# Interpretation

We are 99% confident that the true mean durability of the print-heads (in millions of characters printed before failure) lies between 1.090 million and 1.387 million characters.

# b. Build 99% Confidence Interval Using Known Population Standard Deviation

### If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

To construct a 99% confidence interval for the mean durability of the print-heads, given that the population standard deviation is known to be 0.2 million characters, we can use the z-distribution.

Sample size (n): 15

Sample mean: 1.239 million characters (as calculated previously)

Population standard deviation: 0.2 million characters

Confidence level: 99%

In [10]:
# Known population standard deviation
sigma = 0.2

In [11]:
# z-critical value for 99% confidence interval
z_critical = stats.norm.ppf(0.995)  # 99% confidence level => 0.995 in two-tailed test
z_critical

2.5758293035489004

In [12]:
# Standard Error
SE = sigma / np.sqrt(n)
SE

0.051639777949432225

In [13]:
# Margin of Error
ME = z_critical * SE
ME

0.13301525327090588

In [14]:
# Confidence Interval
CI_lower = mean_x - ME
CI_upper = mean_x + ME
(CI_lower, CI_upper)

(1.1056514133957607, 1.3716819199375725)

# Interpretation

We are 99% confident that the true mean durability of the print-heads lies between 1.106 million and 1.372 million characters.